File: typescript/twoSum.ts
function twoSum(nums: number[], target: number): number[] | null {
  const map = new Map<number, number>()
  for (let i = 0; i < nums.length; i++) {
      const num = nums[i]
      const complement = target - num!
      if (map.has(complement)) {
          return [map.get(complement)!, i]
      }
      map.set(num!, i)
  }
  return null
}
console.log(twoSum([2, 7, 11, 15], 9));

File: golang/twoSum.go
package main

func TwoSum(nums []int, target int) []int {

	map1 := make(map[int]int)

	for i, num := range nums {
		complement := target - num
		if idx, found := map1[complement]; found {
			return []int{idx, i}
		}
		map1[num] = i
	}
	return nil
}
